src/Core/CoreKernel.php

```diff
@@ -116,6 +117,27 @@ protected function getDatabaseConfig()
             "password" => Environment::getEnv('SS_DATABASE_PASSWORD') ?: null,
         ];
 
+        // Only add SSL keys in the array if there is an actual value associated with them
+        $sslConf = [
+            'ssl_key' => 'SS_DATABASE_SSL_KEY',
+            'ssl_cert' => 'SS_DATABASE_SSL_CERT',
+            'ssl_ca' => 'SS_DATABASE_SSL_CA',
+            'ssl_cipher' => 'SS_DATABASE_SSL_CIPHER',
+        ];
+        foreach ($sslConf as $key => $envVar) {
+            $envValue = Environment::getEnv($envVar);
+            if ($envValue) {
+                $databaseConfig[$key] = $envValue;
+            }
+        }
+
+        // Having only the key or cert without the other is bad configuration.
+        if (isset($databaseConfig['ssl_key']) xor isset($databaseConfig['ssl_cert'])) {
```

Comment:
Must have either the CA, or both the key and cert, or all three.
This is mirrored in the original implementation: https://github.com/silverstripe/silverstripe-framework/pull/7233/files#diff-6b84b243cc8c7653ef8cadbb9fa5620f991f7454609aaea3f6527e45c3a5820eR128-R142

Reply:
Can you rewrite without `xor`, it's very uncommon so many people would need to lookup what it means in order to read this code

Reply:
I think `xor` as a concept isn't by any means an unknown - but I'll rewrite this as I don't feel any particular need to keep it. It'll just make the condition a little more convoluted but not to the extent that it would be worth me pushing back on.

Reply:
Done